examples/linear_model/plot_quantile_regression.py

```diff
@@ -11,100 +12,296 @@
 
-a = 5
-y = 10 + 0.5 * x + 10 * (rng.pareto(a, size=x.shape[0]) - 1 / (a - 1))
-ax2.plot(x, y_mean, "k--")
+for quantile, y_pred in predictions.items():
+    plt.plot(X, y_pred, label=f"Quantile: {quantile}")
+
+plt.scatter(
+    x[out_bounds_predictions],
+    y_noisy_normal[out_bounds_predictions],
+    color="black",
+    marker="+",
+    alpha=0.5,
+    label="Outside interval",
+)
+plt.scatter(
+    x[~out_bounds_predictions],
+    y_noisy_normal[~out_bounds_predictions],
+    color="black",
+    alpha=0.5,
+    label="Inside interval",
+)
 
-models = []
+plt.legend()
+plt.xlabel("x")
+plt.ylabel("y")
+_ = plt.title("Quantiles of heteroscedastic Normal distributed target")
+
+# %%
+# Since the noise is still Normally distributed, the true mean and conditional
+# median estimated are close to each other. We observe the effect of having an
+# increasing noise variance on the 5% and 95% quantiles: larger is `x` wider
+# is in the quantiles interval.
+#
+# We can repeat the same experiment using the asymmetric Pareto distributed
+# target.
+quantiles = [0.05, 0.5, 0.95]
+predictions = {}
+out_bounds_predictions = np.zeros_like(y_true, dtype=np.bool_)
 for quantile in quantiles:
     qr = QuantileRegressor(quantile=quantile, alpha=0)
-    qr.fit(X, y)
-    ax2.plot([0, 10], qr.predict([[0], [10]]))
-    models.append(qr)
+    y_pred = qr.fit(X, y_noisy_pareto).predict(X)
+    predictions[quantile] = y_pred
+
+    if quantile == min(quantiles):
+        out_bounds_predictions = np.logical_or(
+            out_bounds_predictions, y_pred >= y_noisy_pareto
+        )
+    elif quantile == max(quantiles):
+        out_bounds_predictions = np.logical_or(
+            out_bounds_predictions, y_pred <= y_noisy_pareto
+        )
+
+# %%
+plt.plot(X, y_true, color="black", linestyle="dashed", label="True mean")
+
+for quantile, y_pred in predictions.items():
+    plt.plot(X, y_pred, label=f"Quantile: {quantile}")
+
+plt.scatter(
+    x[out_bounds_predictions],
+    y_noisy_pareto[out_bounds_predictions],
+    color="black",
+    marker="+",
+    alpha=0.5,
+    label="Outside interval",
+)
+plt.scatter(
+    x[~out_bounds_predictions],
+    y_noisy_pareto[~out_bounds_predictions],
+    color="black",
+    alpha=0.5,
+    label="Inside interval",
+)
+
+plt.legend()
+plt.xlabel("x")
+plt.ylabel("y")
+_ = plt.title("Quantiles of asymmetric Pareto distributed target")
+
 
-plot_points_highlighted(x, y, models[0], models[2], ax2)
-ax2.set_xlabel("x")
-ax2.set_ylabel("y")
-ax2.set_title("Quantiles of asymmetric Pareto distributed target")
-ax2.legend(["true mean"] + quantiles, loc="lower right")
-ax2.yaxis.set_tick_params(labelbottom=True)
+# %%
+# Due to the asymmetry distribution of the noise, we observe that the true mean
+# and estimated conditional median are different. We also observe that each
+# quantile models have different model's parameter to better fit the desired
+# quantile.
```

Comment:
Note that ideally, all quantiles would be parallel in this case, which would become more visible with more data points or less extreme quantiles, e.g. 10% and 90%maybe.